### ISSUE TITLE
[Feature request] Mark video as watched

### ISSUE THREAD

**Issue by y-lee** [yuliskov/SmartTubeNext#72]
First, thank you very much for implementing features such as remembering the playback speed.

A way to quickly mark a video as watched, perhaps from the long press popup, would be welcome. This way I can teach the YouTube algorithm that I am interested in the video's topic and show me videos that others who watched that video watched, without having to watch the video again or manually skip to the end.

If this is not possible (to avoid fraudulent automated views), SmartYouTube has a way of marking a video as watched by pushing Channel Up right after starting playback. The same feature (whether done that way, or by a button on the UI, or both) would be very welcome.

**Comment by gate1975mlm** [comment]
+1

**Comment by Jahfry** [comment]
+1 as well. The only option seems to be doing "Not Interested" but too much of that directly messes up algorithm's ability to find similar videos. YouTube has a second set of options after doing "Not Interested" that includes "Already Watched". This would be the much preferred method.

**Comment by y-lee** [comment]
Right. "Not Interested" isn't the desired answer to my request, because I am interested in the video's topic; I just don't want to watch this one (or, often, watch it again).

**Comment by y-lee** [comment]
If one pushes `BACK` within about one second after playback starts (not during spinner; during actual playback), the video is marked as being fully watched. (If one waits more than one second, the video is marked as partly watched.) Can this be used to implement this feature?
